bears/general/MementoBear.py

```diff
@@ -86,55 +62,40 @@ def run(self, filename, file,
         `do_not_ever_open = 'https://api.acme.inc/delete-all-data'` wiping out
         all your data.
 
-        :param network_timeout:    A dict mapping URLs and timeout to be
-                                   used for that URL. All the URLs that have
-                                   the same host as that of URLs provided
-                                   will be passed that timeout. It can also
-                                   contain a wildcard timeout entry with key
-                                   '*'. The timeout of all the websites not
-                                   in the dict will be the value of the key
-                                   '*'.
-        :param link_ignore_regex:  A regex for urls to ignore.
-        :param link_ignore_list:   Comma separated url globs to ignore.
+        :param dependency_results: Results given by URLBear.
         :param follow_redirects:   Set to true to check all redirect urls.
         """
         self._mc = MementoClient()
 
-        network_timeout = {urlparse(url).netloc
-                           if not url == '*' else '*': timeout
-                           for url, timeout in network_timeout.items()}
-
-        if link_ignore_list != self.DEFAULT_IGNORE:
-            link_ignore_list.extend(self.DEFAULT_IGNORE)
-
-        for (line_number, link,
-             code, context) in self.analyze_links_in_file(
-                file, network_timeout, link_ignore_regex, link_ignore_list):
-            status = MementoBear.check_archive(self._mc, link)
-            if not status:
-                yield Result.from_values(
-                    self,
-                    ('This link is not archived yet, visit '
-                     'https://web.archive.org/save/%s to get it archived.'
-                     % link),
-                    file=filename,
-                    line=line_number,
-                    severity=RESULT_SEVERITY.INFO
-                )
-
-            if follow_redirects and 300 <= code < 400:  # HTTP status 30x
-                redirect_urls = MementoBear.get_redirect_urls(link)
-
-                for url in redirect_urls:
-                    status = MementoBear.check_archive(self._mc, url)
-                    if not status:
-                        yield Result.from_values(
-                            self,
-                            ('This link redirects to %s and not archived yet, '
-                             'visit https://web.archive.org/save/%s to get it '
-                             'archived.'
-                             % (url, url)),
-                            file=filename,
-                            line=line_number,
-                            severity=RESULT_SEVERITY.INFO
-                        )
+        for result in dependency_results.get(URLBear.name, []):
+            line_number, link, code, context = result.contents
+
+            if code and 200 <= code < 400:
```

Comment:
You've changed the indent levels of the code, resulting in lots of lines changing, which makes review harder.
This can be avoided here, by using

`if not (code and 200 <= code < 400): continue`